src/libcollections/format_helpers.rs

```diff
@@ -0,0 +1,257 @@
+        if i != 0 { try!(write!(f, ", ")); }
+        try!(write!(f, "{:x}", e));
+    }
+
+    Result::Ok(())
+}
+
+pub fn seq_fmt_upper_exp<I: Iterator>(s: I, f: &mut Formatter) -> fmt::Result
+    where I::Item:fmt::UpperExp
+{
+    for (i, e) in s.enumerate() {
+        if i != 0 { try!(write!(f, ", ")); }
+        try!(write!(f, "{:E}", e));
+    }
+
+    Result::Ok(())
+}
+
+pub fn seq_fmt_lower_exp<I: Iterator>(s: I, f: &mut Formatter) -> fmt::Result
+    where I::Item:fmt::LowerExp
+{
+    for (i, e) in s.enumerate() {
+        if i != 0 { try!(write!(f, ", ")); }
+        try!(write!(f, "{:e}", e));
+    }
+
+    Result::Ok(())
+}
+
+pub fn map_fmt_display<K, V, I: Iterator<Item=(K, V)>>(s: I, f: &mut Formatter) -> fmt::Result
+    where K:fmt::Display,
+          V:fmt::Display
+{
+    for (i, (k, v)) in s.enumerate() {
+        if i != 0 { try!(write!(f, ", ")); }
+        try!(write!(f, "{}: {}", k, v));
+    }
+
+    Result::Ok(())
+}
+
+pub fn map_fmt_octal<K, V, I: Iterator<Item=(K, V)>>(s: I, f: &mut Formatter) -> fmt::Result
+    where K:fmt::Octal,
+          V:fmt::Octal
+{
+    for (i, (k, v)) in s.enumerate() {
+        if i != 0 { try!(write!(f, ", ")); }
+        try!(write!(f, "{:o}: {:o}", k, v));
+    }
+
+    Result::Ok(())
+}
+
+pub fn map_fmt_binary<K, V, I: Iterator<Item=(K, V)>>(s: I, f: &mut Formatter) -> fmt::Result
+    where K:fmt::Binary,
+          V:fmt::Binary
+{
+    for (i, (k, v)) in s.enumerate() {
+        if i != 0 { try!(write!(f, ", ")); }
+        try!(write!(f, "{:b}: {:b}", k, v));
+    }
+
+    Result::Ok(())
+}
+
+pub fn map_fmt_upper_hex<K, V, I: Iterator<Item=(K, V)>>(s: I, f: &mut Formatter) -> fmt::Result
+    where K:fmt::UpperHex,
+          V:fmt::UpperHex
+{
+    for (i, (k, v)) in s.enumerate() {
+        if i != 0 { try!(write!(f, ", ")); }
+        try!(write!(f, "{:X}: {:X}", k, v));
+    }
+
+    Result::Ok(())
+}
+
+pub fn map_fmt_lower_hex<K, V, I: Iterator<Item=(K, V)>>(s: I, f: &mut Formatter) -> fmt::Result
+    where K:fmt::LowerHex,
+          V:fmt::LowerHex
+{
+    for (i, (k, v)) in s.enumerate() {
+        if i != 0 { try!(write!(f, ", ")); }
+        try!(write!(f, "{:x}: {:x}", k, v));
+    }
+
+    Result::Ok(())
+}
+
+pub fn map_fmt_upper_exp<K, V, I: Iterator<Item=(K, V)>>(s: I, f: &mut Formatter) -> fmt::Result
+    where K:fmt::UpperExp,
+          V:fmt::UpperExp
+{
+    for (i, (k, v)) in s.enumerate() {
+        if i != 0 { try!(write!(f, ", ")); }
+        try!(write!(f, "{:E}: {:E}", k, v));
+    }
+
+    Result::Ok(())
+}
+
+pub fn map_fmt_lower_exp<K, V, I: Iterator<Item=(K, V)>>(s: I, f: &mut Formatter) -> fmt::Result
+    where K:fmt::LowerExp,
+          V:fmt::LowerExp
+{
+    for (i, (k, v)) in s.enumerate() {
+        if i != 0 { try!(write!(f, ", ")); }
+        try!(write!(f, "{:e}: {:e}", k, v));
+    }
+
+    Result::Ok(())
+}
+
+pub fn vec_map_fmt_debug<K, V, I: Iterator<Item=(K, V)>>(s: I, f: &mut Formatter) -> fmt::Result
```

Comment:
I would just change the behaviour. Debug and Display are identical for primitives as far as I know, anyway.